### DISCUSSION TITLE
next-cli: print list of all routes

### DISCUSSION THREAD

**Discussion by landsman** [vercel/next.js#57352] [Ideas]
### Goals

I want to export all pages in terms of URLs from the app. 

It's a quite common command in frameworks.
Output can be just a list of slugs, row by row.

Thats includes:
1. pages
2. rewrites
3. redirects

Each of these is some URL that I have to count that exists in terms of migration or maintenance.

### Background

How can I print it out during the development without doing a full app build? Where are not redirects, or rewrites from next config anyway.

### Proposal

command:
`npx next routes`

**Comment by landsman** [comment]
what do you think @leerob ?

**Comment by pablolizardo** [comment]
Hi! i wrote this tool to solve this https://www.npmjs.com/package/next-list
It will list all your pages and API routes, distinguishing whether they’re dynamic, server, client, etc.

**Reply by pablolizardo** [reply]
`npx next-list` try it @revskill10 ! ;)

**Reply by stuff** [reply]
the tool is very nice, great job!
is it planned to have a version exposing internal functions, in order to use it inside other nodejs code, instead of the cli?

**Reply by PeppyH** [reply]
This tool is exactly what I was looking for, thanks!~
